src/Language/Avro/Parser.hs

```diff
@@ -262,11 +262,15 @@ readWithImports baseDir initialFile = do
   where
     oneOfTwo :: FilePath -> IO (Either (ParseErrorBundle T.Text Char) FilePath)
     oneOfTwo p = do
-      let path1 = baseDir </> p
-          path2 = baseDir </> takeDirectory initialFile </> p
+      let dir = takeDirectory initialFile
+          path1 = baseDir </> p
+          path2 = baseDir </> dir </> p
       options <- (,) <$> doesFileExist path1 <*> doesFileExist path2
       pure $ case options of
         (True, False) -> Right p
-        (False, True) -> Right $ takeDirectory initialFile </> p
+        (False, True) -> Right $ dir </> p
         (False, False) -> runParser (fail $ "Import not found: " ++ p) initialFile ""
-        (True, True) -> runParser (fail $ "Duplicate files found: " ++ p) initialFile ""
+        (True, True) ->
+          if dir == "."
```

Comment:
Also, you should [`normalise`](https://hackage.haskell.org/package/filepath-1.4.2.1/docs/System-FilePath-Posix.html#v:normalise) before dealing with file paths.